str/associatr/multiple_testing_correction/run_gene_level_pval.py

```diff
@@ -0,0 +1,306 @@
+            )
+            for i in range(0, len(gene_files), genes_per_job):
+                batch_gene_files = gene_files[i : i + genes_per_job]
+                j = get_batch('Compute gene level pvals').new_python_job(
+                    name=f'Compute gene-level p-values for genes {i+1}-{i+genes_per_job}'
+                )
+                j.cpu(0.25).memory('lowmem')
+                f = get_batch('Compute gene level pvals').new_python_job(
+                    name=f'Compute gene-level Bonferroni p-values for genes {i+1}-{i+genes_per_job}'
+                )
+                f.cpu(0.25).memory('lowmem')
+                for gene_file in batch_gene_files:
+                    # read the raw results
+                    gene_results = pd.read_csv(gene_file, sep='\t')
+                    pvals = gene_results.iloc[:, 5]  # stored in the 6th column
+                    # Find and store the attributes of the locus with lowest raw pval
+                    # Find the minimum value in column 6
+                    min_value = gene_results.iloc[:, 5].min()
+                    # Find the rows with the minimum value in column 6
+                    min_rows = gene_results[gene_results.iloc[:, 5] == min_value]
+
+                    if len(min_rows) == 1:
+                        chr = [min_rows.iloc[:, 0].values[0]]
+                        pos = [min_rows.iloc[:, 1].values[0]]
+                        n_samples_tested = [min_rows.iloc[:, 3].values[0]]
+                        raw_pval = [min_rows.iloc[:, 5].values[0]]
+                        coeff = [min_rows.iloc[:, 6].values[0]]
+                        se = [min_rows.iloc[:, 7].values[0]]
+                        r2 = [min_rows.iloc[:, 8].values[0]]
+                        motif = [min_rows.iloc[:, 9].values[0]]
+                        ref_len = [min_rows.iloc[:, 11].values[0]]
+                        allele_frequency = [min_rows.iloc[:, 12].values[0]]
+                    else:
+                        chr = []
+                        pos = []
+                        n_samples_tested = []
+                        raw_pval = []
+                        coeff = []
+                        se = []
+                        r2 = []
+                        motif = []
+                        ref_len = []
+                        for _index, row in min_rows.iterrows():
+                            chr.append(row.iloc[0])
+                            pos.append(row.iloc[1])
+                            n_samples_tested.append(row.iloc[3])
+                            raw_pval.append(row.iloc[5])
+                            coeff.append(row.iloc[6])
+                            se.append(row.iloc[7])
+                            r2.append(row.iloc[8])
+                            motif.append(row.iloc[9])
+                            ref_len.append(row.iloc[11])
+                            allele_frequency.append(row.iloc[12])
```

Comment:
Is there a reason for this? Wouldn't iterating over a 1-length object also be valid, so you can loop over the rows even if there's only a single row?

Reply:
yup good point